### PR TITLE
Log1p

### LINKED ISSUE #16519: log1p
This is created for to added log1p in paddle.tensor.tensor.Tensor from issue #15115

### PR DESCRIPTION
**Pull request by MuhammadNizamani**
Adding log1p function paddle.tensor.tensor.Tensor from issue Closed  #16519

### CODE REVIEW

**Comment by MuhammadNizamani** [comment]
@zaeemansari70  sir waiting for your review.

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @MuhammadNizamani,
Sorry for the delay in the review process. Thanks a lot for your patience. :slightly_smiling_face: 

I tested your changes locally they are failing for torch, with 
` AssertionError:  the results from backend torch and ground truth framework paddle do not match
E            inf!=87.49822998046875 `
The array being passed has the value of `1.0000001e+38`, maybe try looking into applying a small safety factor and see if it fixes it!

Let me know if you have any questions :slightly_smiling_face: 
Thanks!

**Comment by MuhammadNizamani** [comment]
@zaeemansari70 can you tell how can I test for torch.
and another question this paddle is tensor base then why you are runing it in torch.

**Review by zaeemansari70 — COMMENTED** [review]
HI @MuhammadNizamani,
Just for the context, when we run a frontend function, our  `ground_truth` is the actual `frontend` function from the original framework and is tested against every backend which we have in our `Ivy` for that specific function.
There can be an issue in the sometimes die to which the tests fail but this time is particularly related to the frontends as the array which is being passed inside the function is a huge number which can be fixed by a safety factor in your test function.

I've also added the falsifying example so you can test on it!
```
E           AssertionError:  the results from backend torch and ground truth framework paddle do not match
E            inf!=87.49822998046875 
E           
E           
E           Falsifying example: test_paddle_log1p(
E               frontend='paddle',
E               on_device='cpu',
E               dtype_and_x=(['float32'], [array(1.0000001e+38, dtype=float32)]),
E               frontend_method_data=FrontendMethodData(ivy_init_module=<module 'ivy.functional.frontends.paddle' from '/workspaces/ivy/ivy/functional/frontends/paddle/__init__.py'>, framework_init_module=<module 'paddle' from '/opt/fw/paddle/paddle/__init__.py'>, init_name='to_tensor', method_name='log1p'),
E               init_flags=FrontendMethodTestFlags(
E                   num_positional_args=1,
E                   as_variable=[False],
E                   native_arrays=[False],
E               ),
E               method_flags=FrontendMethodTestFlags(
E                   num_positional_args=0,
E                   as_variable=[False],
E                   native_arrays=[False],
E               ),
E           )
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.78.3', b'AAEAAQAAAIfiztMwAAAAAAAAAAAAAAA=') as a decorator on your test case

Thanks! :slightly_smiling_face:

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 1]
```diff
@@ -1285,3 +1285,47 @@ def test_paddle_bitwise_xor(
...
+
+# log1p
+SAFETY_FACTOR = 0.5  # Adjust the safety factor as per your requirements
+
+
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="log1p",
+    dtype_and_x=helpers.dtype_and_values(
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1298]: The safety factor should be added in the data generation, you have to use the argument provided by `dtype_and_values()`
You can find multiple examples in the code base where we use it, you can take reference from there :+1:

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @MuhammadNizamani ,

Requested some changes :slightly_smiling_face: 

Thanks!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py (hunk 1 above)
Comment (MuhammadNizamani) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1298]: Ok sir

**Comment by MuhammadNizamani** [comment]
@zaeemansari70 check the changes

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 2]
```diff
@@ -1380,3 +1380,37 @@ def test_paddle_rsqrt(
...
     )
+
+
+# log1p
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="log1p",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1391]: Kindly try to use either the "log" safety factor with a number of like `12` or something, a simple safety factor of `2` does not make any difference, the test is still failing for torch for `input=[array(1.e+307)]`

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 3]
```diff
@@ -208,6 +208,10 @@ def allclose(self, other, rtol=1e-05, atol=1e-08, equal_nan=False, name=None):
     def sort(self, axis=-1, descending=False, name=None):
         return ivy.sort(self._ivy_array, axis=axis, descending=descending)
 
+    @with_unsupported_dtypes({"2.4.2 and below": ("float16", "bfloat16")}, "paddle")
```
Comment (zaeemansari70) [inline ivy/functional/frontends/paddle/tensor/tensor.py:211]: The version should be `2.5.0` else the test fails

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
HI @MuhammadNizamani 

I have requested some changes, kindly make sure that my your code passes the test cases locally on your system. This cannot be merged if the tests fail :slightly_smiling_face: 

Feel free to ask any questions, thanks :slightly_smiling_face:

**Review by zaeemansari70 — COMMENTED** [review]
The changes look good to me :+1: 
Kindly resolve the conflicts so I can merge the PR. :slightly_smiling_face:

**Comment by MuhammadNizamani** [comment]
On it

**Comment by zaeemansari70** [comment]
Hi @MuhammadNizamani ,

The conflicts are still being showed here :thinking:

**Review by zaeemansari70 — APPROVED** [review]
There was a small change which needed to be made, now your PR looks good to be merged, thanks! :slightly_smiling_face:

**Pull request merged**